### ISSUE TITLE
attention mask 有bug ？

### ISSUE THREAD

**Issue by masonwang513** [THUDM/GLM-4-Voice#182]
![Image](https://github.com/user-attachments/assets/2222bffe-21c4-4971-978f-5d4a4d03eb9a)

红色框内的代码是不是有bug?  如果是为了屏蔽query本身是padding tokens，不要attend任何key,  -= 不起任何作用， 应该改为如下+=才对啊 
if not past_length and padding_mask is not None:
            full_attention_mask += padding_mask.unsqueeze(-1) - 1
full_attention_mask = (full_attention_mask < 0.5).bool()
